app.js

```diff
@@ -2,22 +2,120 @@
   if (!window.addEventListener) return // Check for IE9+
 
   let options = INSTALL_OPTIONS
+  let updateTimeout
+  const delay = 1500
+  const elements = []
+  const PAYPAL_SCRIPT_URL = "https://cdn.rawgit.com/paypal/JavaScriptButtons/master/dist/button.js"
+  const TIME_PERIOD_SYMBOLS = {
+    D: "Daily",
+    W: "Weekly",
+    M: "Monthly",
+    Y: "Yearly"
+  }
+
+  function updateElements() {
+    if (options.merchant) {
+      const {buttons} = options
+
+      buttons
+        .filter($ => $.name && $.amount)
+        .forEach((attrs, i) => {
+          const paypalButton = document.createElement("script")
+          const paypalInfoWrapper = document.createElement("eager-paypal-info-wrapper")
+          const paypalItemName = document.createElement("eager-paypal-item-name")
+          const paypalPrice = document.createElement("eager-paypal-price")
+
+          let currencySymbol
+          let times
+
+          if (options.region.currency === "GBP") {
+            currencySymbol = "£"
+          }
+          else {
+            currencySymbol = "$"
+          }
+
+          paypalItemName.innerHTML = `${attrs.name}`
+
+          paypalPrice.innerHTML = `${currencySymbol}${attrs.amount} ${options.region.currency} `
 
-  function updateElement() {
+          if (attrs.type === "subscribe") {
+            if (attrs.recurrence === 1){
+              times = "time"
+            }
+            else {
+              times = "times"
+            }
+            paypalPrice.innerHTML += `${attrs.recurrence} ${times} ${TIME_PERIOD_SYMBOLS[attrs.timePeriod]} `
+          }
+
+          if (options.region.tax && !attrs.shipping) {
+            paypalPrice.innerHTML += `<small>+ ${currencySymbol}${options.region.tax} tax </small>`
+          }
+          else if (attrs.shipping && !options.region.tax) {
+            paypalPrice.innerHTML += `<small>+ ${currencySymbol}${attrs.shipping} shipping</small>`
+          }
+          else if (attrs.shipping && options.region.tax) {
+            const shippingAndTax = attrs.shipping + options.region.tax
+
+            paypalPrice.innerHTML += `<small>+ ${currencySymbol}${shippingAndTax} shipping and tax</small>`
+          }
+
+          // TODO find production host
+          paypalButton.src = `${PAYPAL_SCRIPT_URL}?merchant=${options.merchant}`
+          paypalButton.async
+          paypalButton.setAttribute("data-button", attrs.type)
+          paypalButton.setAttribute("data-type", "form")
+          paypalButton.setAttribute("data-name", attrs.name)
+          if (attrs.type === "donate"){
+            paypalButton.setAttribute("data-amount-editable", 100)
```

Comment:
Unexpected [Magic Number](http://c2.com/cgi/wiki?MagicNumber) 
What does this do? Why 100?